src/providers/mssql/qgsmssqldataitems.cpp

```diff
@@ -270,24 +251,29 @@ QVector<QgsDataItem *> QgsMssqlConnectionItem::createChildren()
       }
     }
 
-    // add missing schemas (i.e., empty schemas)
-    const QString uri = connInfo();
-    const QStringList allSchemas = QgsMssqlConnection::schemas( uri, nullptr );
-    for ( const QString &schema : allSchemas )
+    if ( !mUseGeometryColumns )
     {
-      if ( addedSchemas.contains( schema ) ||
-           ( schemaSettings.contains( schema ) && !schemaSettings.value( schema ).toBool() ) )
-        continue;
+      // add missing schemas (i.e., empty schemas)
+      const QString uri = connInfo();
+      const QStringList allSchemas = QgsMssqlConnection::schemas( uri, nullptr );
+      QVariantMap schemaSettings = mSchemaSettings.value( mDatabase ).toMap();
+      for ( const QString &schema : allSchemas )
+      {
+        if ( addedSchemas.contains( schema ) ||
+             ( schemaSettings.contains( schema ) && !schemaSettings.value( schema ).toBool() ) )
+          continue;
```

Comment:
This code for filtering should not be needed anymore if filtering is done on the server... or did I get something wrong here?

Reply:
here, there is a new query that returns ALL the schema to search which are not already added. if we do not do another filtering the empty one not selected will appears in the browser.
It could be possible to filter, as well here, on the server side, but we have to modify `QgsMssqlConnection::schemas()` method

Reply:
Ah right, I just got confused...

May I suggest to split that `if` into two separate ifs - I always find it difficult to parse these combined ifs with multiple and, or, not clauses :-)

... and shouldn't we have the condition to use `mSchemasFilteringEnabled` instead of `schemaSettings.contains( schema )` ? With the latter, we could be showing a schema item for one which was recently added, which is not in the list of filtered schemas, but there would not be any tables listed with it...?

so my suggestion would be to change it to this:
```
if ( mSchemasFilteringEnabled && !schemaSettings.value( schema ).toBool() )
  continue;  // user does not want it to be shown

if ( addedSchemas.contains( schema ) )
  continue;
```